### PR TITLE
fix(ignored-posts): Filter wp_count_posts() ...

### PR DESCRIPTION
**Pull request by Camwyn**
…to include a dummy count for tribe-ignored status to prevent notices.

By making the status available in `show_in_admin_status_list` we tell WP to handle it for every post table - but WP doesn't check to ensure it's actually there.
So it tries to access a non-existent property and throws a notice. This hooks into `wp_count_posts()` and adds a dummy (0) value for it to prevent notices.

### CODE REVIEW

File: src/Tribe/Ignored_Events.php [hunk 1]
```diff
@@ -975,7 +977,32 @@ public function hook() {
...
+
+		/**
+		 * Patch post-list-table queries to include ignored status,
+		 * as WP just accesses the stati without checking if they exist.
+		 *
+		 * @see wp-admin/includes/class-wp-posts-list-table.php->get_views()
+		 *
+		 * @since TBD
+		 *
+		 * @param [type] $counts
```
Comment (tr1b0t) [inline src/Tribe/Ignored_Events.php:993]: :no_entry_sign: Missing parameter comment

<sub>hash: 26872071e95ed082da757a34265be406</sub>

File: src/Tribe/Ignored_Events.php [hunk 2]
```diff
@@ -975,7 +977,32 @@ public function hook() {
...
+		 * @param [type] $type
```
Comment (tr1b0t) [inline src/Tribe/Ignored_Events.php:994]: :no_entry_sign: Missing parameter comment

<sub>hash: ad873851b17c27e4addb44fef684dc8e</sub>

File: src/Tribe/Ignored_Events.php [hunk 3]
```diff
@@ -975,7 +977,32 @@ public function hook() {
...
+		 * @param [type] $perm
```
Comment (tr1b0t) [inline src/Tribe/Ignored_Events.php:995]: :no_entry_sign: Missing parameter comment

<sub>hash: b3b4a11d4ef8adbfa8f9ebf0cc78ebdd</sub>

File: src/Tribe/Ignored_Events.php [hunk 4]
```diff
@@ -975,7 +977,32 @@ public function hook() {
...
+		 * @return void
```
Comment (tr1b0t) [inline src/Tribe/Ignored_Events.php:996]: :no_entry_sign: Function return type is void, but function contains return statement

<sub>hash: 860fb734d76be795bec86ae012c77636</sub>

File: src/Tribe/Ignored_Events.php [hunk 5]
```diff
@@ -975,7 +977,35 @@ public function hook() {
...
+
+		/**
+		 * Patch post-list-table queries to include ignored status,
+		 * as WP just accesses the stati without checking if they exist.
+		 *
+		 * @see wp-admin/includes/class-wp-posts-list-table.php->get_views()
+		 *
+		 * @since TBD
+		 *
+		  * @param object $counts       An object containing the current post_type's post
```
Comment (tr1b0t) [inline src/Tribe/Ignored_Events.php:993]: * :warning: Found precision alignment of 1 spaces.
* :no_entry_sign: Expected 9 space(s) before asterisk; 10 found

<sub>hash: 5b73f0ece26cc3422599e45ae84b2c7a</sub>

Comment (Camwyn) [inline src/Tribe/Ignored_Events.php:993]:
```suggestion
		 * @param object $counts       An object containing the current post_type's post
```

File: src/Tribe/Ignored_Events.php [hunk 6]
```diff
@@ -975,7 +977,35 @@ public function hook() {
...
+		 * @param object $counts       An object containing the current post_type's post
+		 *                              counts by status.
+		 * @param string $unused_type   Post type.
+		 * @param string $unused_perm   The permission to determine if the posts are 'readable'
+		 *                              by the current user.
+		 * @return object $counts       The modified object containing the current post_type's post
+		 *                              counts by status.
+		 */
+		public function patch_count_posts( $counts, $unused_type, $unused_perm ) {
+			$status = $this::$ignored_status;
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:1002]: Use `self::$ignored_status` here.

File: src/Tribe/Ignored_Events.php [hunk 7]
```diff
@@ -975,7 +977,35 @@ public function hook() {
 			 */
 			tribe_notice( 'legacy-ignored-events', array( $this, 'render_notice_legacy' ), 'dismiss=1&type=warning' );
 
+			add_filter( 'wp_count_posts', [ $this, 'patch_count_posts' ], 10, 3 );
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:980]: Remove the last argument, if we only use 1, then let's only require 1.

File: src/Tribe/Ignored_Events.php [hunk 8]
```diff
@@ -975,7 +977,35 @@ public function hook() {
...
+		 *
+		 * @param object $counts       An object containing the current post_type's post
+		 *                              counts by status.
+		 * @param string $unused_type   Post type.
+		 * @param string $unused_perm   The permission to determine if the posts are 'readable'
+		 *                              by the current user.
+		 * @return object $counts       The modified object containing the current post_type's post
+		 *                              counts by status.
+		 */
+		public function patch_count_posts( $counts, $unused_type, $unused_perm ) {
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:1001]: Remove the two unused arguments `$unused_type` and `$unused_perm`.

File: src/Tribe/Ignored_Events.php [hunk 9]
```diff
@@ -975,7 +977,35 @@ public function hook() {
...
+		 * Patch post-list-table queries to include ignored status,
+		 * as WP just accesses the stati without checking if they exist.
+		 *
+		 * @see wp-admin/includes/class-wp-posts-list-table.php->get_views()
+		 *
+		 * @since TBD
+		 *
+		 * @param object $counts       An object containing the current post_type's post
+		 *                              counts by status.
+		 * @param string $unused_type   Post type.
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:995]: Remove doc entry for the two unused args.

File: src/Tribe/Ignored_Events.php [hunk 10]
```diff
@@ -975,7 +977,35 @@ public function hook() {
...
+		 * @param string $unused_perm   The permission to determine if the posts are 'readable'
+		 *                              by the current user.
+		 * @return object $counts       The modified object containing the current post_type's post
+		 *                              counts by status.
+		 */
+		public function patch_count_posts( $counts, $unused_type, $unused_perm ) {
+			$status = $this::$ignored_status;
+
+			if ( ! isset( $counts->$status ) ) {
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:1004]: Would we want to check for an positive integer `>= 0` here; the input should not be trusted here.

Comment (Camwyn) [reply src/Tribe/Ignored_Events.php:1004]: Why? This is just to check for situations when `$counts->$status` is not set and put in a placeholder value - I'm specifically using `isset()` so I don't change any existing values.

Comment (lucatume) [reply src/Tribe/Ignored_Events.php:1002]: Right, thanks.

File: src/Tribe/Ignored_Events.php [hunk 11]
```diff
@@ -975,7 +977,33 @@ public function hook() {
 			 */
 			tribe_notice( 'legacy-ignored-events', array( $this, 'render_notice_legacy' ), 'dismiss=1&type=warning' );
 
+			add_filter( 'wp_count_posts', [ $this, 'patch_count_posts' ], 10 );
```
Comment (lucatume) [inline src/Tribe/Ignored_Events.php:980]: Since you're using the default priority, you can remove it.

Comment (Camwyn) [reply src/Tribe/Ignored_Events.php:980]:
```suggestion
			add_filter( 'wp_count_posts', [ $this, 'patch_count_posts' ] );
```

**Pull request merged**